File: Jack/Assets/HalloweenRange/_Scripts/C#/MuzzleFlashAnimate.cs
using UnityEngine;
using System.Collections;

public class MuzzleFlashAnimate : MonoBehaviour 
{
	public float scaleParam = 0.5f	;			 	// Scale parameter
	public Vector3 scaleVector = Vector3.one;  		// In which direction is muzzle scaling 
	
	//============================================================================================
	void Start()
	{
		SetVisibility (false);
	}
	
	//----------------------------------------------------------------------------------------------
	// Animate muzzle	
	void Update () 
	{
		transform.localScale = scaleVector * Random.Range(scaleParam, scaleParam*3);
		float tempZ = Random.Range(0.0f,90.0f);
		Vector3 temp = new Vector3(transform.localEulerAngles.x, transform.localEulerAngles.y, tempZ);
		transform.localEulerAngles = temp;
		if (light) 
			light.enabled = false;
	}
	//----------------------------------------------------------------------------------------------
	
	void SetVisibility (bool isVisible) 
	{
		renderer.enabled = isVisible;
		if (light) 
			light.enabled = isVisible;
	}
	
	//----------------------------------------------------------------------------------------------
}

File: Jack/Assets/HalloweenRange/_Scripts/Js/MuzzleFlashAnimate.js
//MuzzleFlashAnimate
#pragma strict
var scaleParam: float = 0.5;			 // Scale parameter
var scaleVector: Vector3 = Vector3.one;  // In which direction is muzzle scaling 

//============================================================================================

function Start()
{
  SetVisibility (false);
}

//----------------------------------------------------------------------------------------------
// Animate muzzle

function Update () {
	transform.localScale = scaleVector * Random.Range(scaleParam, scaleParam*3);
	transform.localEulerAngles.z = Random.Range(0,90.0);
	 if (light) light.enabled = false;
}

//----------------------------------------------------------------------------------------------

function SetVisibility (isVisible : boolean) {
    renderer.enabled = isVisible;
    if (light) light.enabled = isVisible;
}

//----------------------------------------------------------------------------------------------